ctru-rs/examples/camera-image.rs

```diff
@@ -0,0 +1,205 @@
+use ctru::console::Console;
+use ctru::gfx::{Screen, Side};
+use ctru::services::cam::{
+    Cam, CamContext, CamOutputFormat, CamPort, CamSelect, CamShutterSoundType, CamSize,
+};
+use ctru::services::hid::KeyPad;
+use ctru::services::{Apt, Hid};
+use ctru::Gfx;
+
+const WIDTH: i16 = 400;
+const HEIGHT: i16 = 240;
+const SCREEN_SIZE: u32 = 192000; // WIDTH * HEIGHT * 2;
+const BUF_SIZE: usize = 384000; // SCREEN_SIZE * 2;
+
+const WAIT_TIMEOUT: i64 = 300000000;
+
+fn main() {
+    ctru::init();
+
+    let apt = Apt::init().expect("Failed to initialize Apt service.");
+    let hid = Hid::init().expect("Failed to initialize Hid service.");
+    let gfx = Gfx::init().expect("Failed to initialize GFX service.");
+
+    gfx.top_screen.borrow_mut().set_double_buffering(true);
+    gfx.bottom_screen.borrow_mut().set_double_buffering(false);
+
+    let _console = Console::init(gfx.bottom_screen.borrow_mut());
+
+    let mut keys_down;
+
+    println!("Initializing camera");
+
+    let mut cam = Cam::init().expect("Failed to initialize CAM service.");
+
+    cam.set_size(
+        CamSelect::SELECT_OUT1_OUT2,
+        CamSize::SIZE_CTR_TOP_LCD,
+        CamContext::CONTEXT_A,
+    )
+    .expect("Failed to set camera size");
+    cam.set_output_format(
+        CamSelect::SELECT_OUT1_OUT2,
+        CamOutputFormat::OUTPUT_RGB_565,
+        CamContext::CONTEXT_A,
+    )
+    .expect("Failed to set camera output format");
+
+    cam.set_noise_filter(CamSelect::SELECT_OUT1_OUT2, true)
+        .expect("Failed to enable noise filter");
+    cam.set_auto_exposure(CamSelect::SELECT_OUT1_OUT2, true)
+        .expect("Failed to enable auto exposure");
+    cam.set_auto_white_balance(CamSelect::SELECT_OUT1_OUT2, true)
+        .expect("Failed to enable auto white balance");
+
+    cam.set_trimming(CamPort::PORT_CAM1, false)
+        .expect("Failed to disable trimming for Cam Port 1");
+    cam.set_trimming(CamPort::PORT_CAM2, false)
+        .expect("Failed to disable trimming for Cam Port 2");
+
+    let mut buf = [0u8; BUF_SIZE];
+
+    gfx.flush_buffers();
+    gfx.swap_buffers();
+    gfx.wait_for_vblank();
+
+    println!("\nPress R to take a new picture");
+    println!("Press Start to exit to Homebrew Launcher");
+
+    while apt.main_loop() {
+        hid.scan_input();
+        keys_down = hid.keys_down();
+
+        if keys_down.contains(KeyPad::KEY_START) {
+            break;
+        }
+
+        if keys_down.bits() & KeyPad::KEY_R.bits() != 0 {
```

Comment:
I think this is a question we should ask for all the examples. The issue with using `KeyPad::contains` or for any other bitflag is if I do any combination of buttons that simply contains the one specified, it will return true. 

For example, if I press A + START the application will exit because it contains the flag. For accuracy, I think masking the `keys_down` with the bits of the target button would be better.

Reply:
There is `.intersect` for that, which is a mask that returns a mask.

For example, to have A + START:
`!keys_down.intersect(KeyPad::KEY_A).intersect(KeyPad::KEY_START).is_empty()`

It is more verbose, but it's functionally the same and more readable. It's up to personal reference, but I don't find it problematic. Also, no examples use button combos right now, so I don't see why we should change the current source code in any case.

Reply:
You can just do unions for that.

Example:
`keys_down.contains(KeyPad::KEY_A | KeyPad::KEY_START)`

No examples use button combos though, so I still find no issue with just `.contains`.

Reply:
Agreed but examples should probably infer intention as the picture in the example is meant to be take when R is pressed not any combination of keys that contains R.

Reply:
Fair enough. I also sort of completely forgot they could just be compared since my brain thinking "bitflag". I'll change it back to contains for this example but I think the `buttons` example could be revisited then because it does use a bitmask to check for a single button and that's also what I was going off of.